### PR TITLE
[CAS-323] Update project dependencies

### PR DESCRIPTION
**Pull request by zsmb13**
https://stream-io.atlassian.net/browse/CAS-323

### Description

Updates project dependencies to the latest versions. See inline comments for more details.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] ~Comparison screenshots added for visual changes~
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-client-sample/build.gradle [hunk 1]
```diff
@@ -78,6 +78,7 @@ dependencies {
     kapt Dependencies.roomCompiler
 
     // Firebase
+    implementation platform(Dependencies.firebaseBom)
```
Comment (zsmb13) [inline stream-chat-android-client-sample/build.gradle:81]: Using the BoM is the recommended way of keeping Firebase dependencies in sync, as their compatible versions have very different version numbers https://firebase.google.com/docs/android/learn-more#bom

File: stream-chat-android-client/src/test/java/io/getstream/chat/android/client/testing/utils.kt [hunk 2]
```diff
@@ -13,7 +13,7 @@ private val resClass = ResClass()
  * [path] "/model.member.json"
  */
 internal fun loadResource(path: String): String {
-    return convert(resClass.javaClass.getResourceAsStream(path))
+    return convert(requireNotNull(resClass.javaClass.getResourceAsStream(path)))
```
Comment (zsmb13) [inline stream-chat-android-client/src/test/java/io/getstream/chat/android/client/testing/utils.kt:16]: AGP 4.1 enforces `@RecentlyNullable` annotations in a stricter way, hence these kinds of changes.

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClientImpl.kt [hunk 3]
```diff
@@ -428,7 +428,7 @@ internal class ChatClientImpl(
     }
 
     override fun getVersion(): String {
-        return BuildConfig.VERSION_NAME + "-" + BuildConfig.BUILD_TYPE
+        return BuildConfig.STREAM_CHAT_CLIENT_VERSION + "-" + BuildConfig.BUILD_TYPE
```
Comment (zsmb13) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClientImpl.kt:431]: Library projects no longer generate `VERSION_CODE` and `VERSION_NAME` values in the `BuildConfig` ([see AGP 4.1 release notes](https://developer.android.com/studio/releases/gradle-plugin#version_properties_removed_from_buildconfig_class_in_library_projects)). These values were often incorrect, and overwritten by manifest merges.

The solution for programatically accessing the values is to add them to the `BuildConfig` for ourselves (see the changes in the `build.gradle` files).

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/ChatImpl.java [hunk 4]
```diff
@@ -148,7 +148,7 @@ public ChatMarkdown getMarkdown() {
     @NotNull
     @Override
     public String getVersion() {
-        return BuildConfig.BUILD_TYPE + ":" + BuildConfig.VERSION_NAME;
+        return BuildConfig.STREAM_CHAT_UI_VERSION + "-" + BuildConfig.BUILD_TYPE;
```
Comment (zsmb13) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/ChatImpl.java:151]: Updated the version string format to match the format in the other two libraries

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/utils/exomedia/core/source/MediaSourceProvider.java [hunk 5]
```diff
@@ -39,11 +39,11 @@
  */
 @SuppressWarnings("WeakerAccess")
 public class MediaSourceProvider {
-    protected static final String USER_AGENT_FORMAT = "ExoMedia %s (%d) / Android %s / %s";
+    protected static final String USER_AGENT_FORMAT = "ExoMedia %s / Android %s / %s";
 
     @NonNull
     @SuppressLint("DefaultLocale")
-    protected String userAgent = String.format(USER_AGENT_FORMAT, BuildConfig.VERSION_NAME, BuildConfig.VERSION_CODE, Build.VERSION.RELEASE, Build.MODEL);
+    protected String userAgent = String.format(USER_AGENT_FORMAT, BuildConfig.STREAM_CHAT_UI_VERSION, Build.VERSION.RELEASE, Build.MODEL);
```
Comment (zsmb13) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/utils/exomedia/core/source/MediaSourceProvider.java:46]: As I understand, the user agent here is an arbitrary string, and the version name should be enough to identify the library version in logs and such.

File: buildSrc/src/main/kotlin/com/getstream/sdk/chat/Dependencies.kt [hunk 6]
```diff
@@ -132,27 +134,25 @@ object Dependencies {
...
     const val navigationFragmentKTX = "androidx.navigation:navigation-fragment-ktx:$NAVIGATION_VERSION"
     const val navigationSafeArgsGradlePlugin = "androidx.navigation:navigation-safe-args-gradle-plugin:$NAVIGATION_VERSION"
     const val navigationRuntimeKTX = "androidx.navigation:navigation-runtime-ktx:$NAVIGATION_VERSION"
     const val navigationUIKTX = "androidx.navigation:navigation-ui-ktx:$NAVIGATION_VERSION"
     const val ok2curl = "com.github.mrmike:ok2curl:$OK2CURL_VERSION"
-    const val okhttp = "com.squareup.okhttp3:okhttp:$OKHTTP_VERSION"
-    const val okhttpLoggingInterceptor = "com.squareup.okhttp3:logging-interceptor:$OKHTTP_VERSION"
-    const val okhttpMockWebserver = "com.squareup.okhttp3:mockwebserver:$OKHTTP_VERSION"
+    const val okhttp = "com.squareup.okhttp3:okhttp"
+    const val okhttpBom = "com.squareup.okhttp3:okhttp-bom:$OKHTTP_BOM_VERSION"
```
Comment (zsmb13) [inline buildSrc/src/main/kotlin/com/getstream/sdk/chat/Dependencies.kt:152]: Same as with Firebase, started using the BoM for OkHttp too, as they do publish one

**Review by samiuelson — DISMISSED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,12 +1,17 @@
 # To be released:
-## stream-chat-anddroid
+
+## All libraries
```
Comment (samiuelson) [inline CHANGELOG.md:3]: Personally, I'd prefer something like this instead
```suggestion
## Common / project-wide changes 
```

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**